src/doc/grammar.md

```diff
@@ -512,7 +512,7 @@ unit_expr : "()" ;
 ```antlr
 struct_expr : expr_path '{' ident ':' expr
                       [ ',' ident ':' expr ] *
-                      [ ".." expr ] '}' |
+                      [ ".." expr ] ? '}' |
```

Comment:
Usually in BNFR () is used for grouping and [] for optional but in grammar.md it's used just for grouping instead of (). ( [quote](https://github.com/rust-lang/rust/blob/master/src/doc/grammar.md#notation): `Square brackets are used to group rules.`